integration/spark/app/src/test/java/io/openlineage/spark/agent/OpenLineageSparkListenerTest.java

```diff
@@ -131,15 +131,25 @@ void testOpenlineageDisableDisablesExecution() throws URISyntaxException {
       when(Environment.getEnvironmentVariable("OPENLINEAGE_DISABLED")).thenReturn("true");
 
       ContextFactory contextFactory = mock(ContextFactory.class);
+      when(OpenLineageSparkListener.getEventProcessingTimeoutSeconds())
+          .thenReturn(Optional.empty());
 
-      OpenLineageSparkListener.init(contextFactory);
-      OpenLineageSparkListener listener = new OpenLineageSparkListener();
+      try (MockedStatic mockedListener = mockStatic(OpenLineageSparkListener.class)) {
+        mockedListener
+            .when(OpenLineageSparkListener::getEventProcessingTimeoutSeconds)
+            .thenReturn(Optional.empty());
+        when(OpenLineageSparkListener.getEventProcessingTimeoutSeconds())
+            .thenReturn(Optional.empty());
 
-      listener.onJobStart(
-          new SparkListenerJobStart(
-              0, 2L, ScalaConversionUtils.<StageInfo>asScalaSeqEmpty(), new Properties()));
+        OpenLineageSparkListener.init(contextFactory);
+        OpenLineageSparkListener listener = new OpenLineageSparkListener();
 
-      verify(contextFactory, never()).createSparkSQLExecutionContext(anyLong());
+        listener.onJobStart(
+            new SparkListenerJobStart(
+                0, 2L, ScalaConversionUtils.<StageInfo>asScalaSeqEmpty(), new Properties()));
+
+        verify(contextFactory, never()).createSparkSQLExecutionContext(anyLong());
+      }
```

Comment:
Is it doable to write a unit test that verifies the behaviour? 

Possible test scenario:
 - create a context that for methods like `context.end(jobEnd);` that sleeps `100ms` 
 - configure timeout to `10ms`
 - verify execution lasting less than 100ms